### PR TITLE
Modeling users

### PR DESCRIPTION
**Pull request by kalpeshDocFox**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  before_save {  email.downcase! }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 }, format: { with: VALID_EMAIL_REGEX }, uniqueness: true,uniqueness: { case_sensitive: false }
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }
+end
```
Comment (c13e) [inline app/models/user.rb:20]: I'll ask @oratilemlambo to add you to her repo so you can have a look at the comments I made there on [the same PR](https://github.com/oratilemlambo/sample_app/pull/4), that way I don't have to repeat myself :)

File: test/controllers/static_pages_controller_test.rb [hunk 2]
```diff
@@ -1,21 +1,21 @@
 require 'test_helper'
 class StaticPagesControllerTest < ActionDispatch::IntegrationTest
-  test "should get home" do
+  test "GET #home should render the root page, a success response, and a correct title " do
```
Comment (c13e) [inline test/controllers/static_pages_controller_test.rb:3]: great test names 💯

File: test/models/user_test.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  def setup
+    @user = User.new(name: "Example User", email: "user@example.com", password: "foobar", password_confirmation: "foobar")
```
Comment (c13e) [inline test/models/user_test.rb]: this line is starting to get too long which negatively affects readability.

I'd suggest splitting it into multiple lines, ex: 
```rb
    @user = User.new(
      name: "Example User",
      ...
    )
```

You can using a formatter to automate this in your editor. We use [Rubocop](https://github.com/rubocop/rubocop) in the main Docfox repo and there are others like Rufo.

Something I also find useful is to add a visual indicator at the 80-character line which helps me decide when to start breaking up a long line into multiple lines. For instance in VS Code you can get this by adding "editor.rulers": [80] to the settings (Cmd+,) but other editors will have a similar feature

File: test/models/user_test.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  test "name should not be too long" do
+    @user.name = "a" * 51
```
Comment (c13e) [inline test/models/user_test.rb]: ditto, see comments I made on Oratile's PR re: test names + extracting & using constants for these constraint values like max name length

File: test/models/user_test.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    assert_not @user.valid?
+  end
+  test "email should not be too long" do
```
Comment (c13e) [inline test/models/user_test.rb]:
```suggestion
  end
  
  test "email should not be too long" do
```
add a new line between your test cases

File: test/models/user_test.rb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    @user.email = "a" * 244 + "@example.com"
+    assert_not @user.valid?
+  end
+  test "email validation should accept valid addresses" do
+    valid_addresses = %w[user@example.com USER@foo.COM A_US-ER@foo.bar.org first.last@foo.jp alice+bob@baz.cn]
```
Comment (c13e) [inline test/models/user_test.rb]: ditto re: formatting long lines

File: test/models/user_test.rb [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  end
+  test "password should be present (nonblank)" do
+    @user.password = @user.password_confirmation = " " * 6
+    assert_not @user.valid?
+  end
+  test "password should have a minimum length" do
+    @user.password = @user.password_confirmation = "a" * 5
+    assert_not @user.valid?
+  end
+end
```
Comment (c13e) [inline test/models/user_test.rb:56]: see if you can configure your editor to automatically add a new line at the end of each file to eliminate this missing new line icon, for ex: in VS Code it's adding `"files.insertFinalNewline": true,` to settings file

**Review by c13e — APPROVED** [review]
Noice!

File: app/models/user.rb (hunk 1 above)
Comment (kalpeshDocFox) [reply app/models/user.rb:20]: Thanks Reviewed the changes on her pr swell

**Pull request merged**